packages/ui-drawer-layout/src/DrawerLayout/DrawerContent/index.tsx

```diff
@@ -37,124 +38,109 @@ import generateComponentTheme from './theme'
 import { propTypes, allowedProps } from './props'
 import type {
   DrawerLayoutContentProps,
-  DrawerLayoutContentState
+  DrawerLayoutContentStyleProps
 } from './props'
 
 /**
 ---
 parent: DrawerLayout
 id: DrawerLayout.Content
 ---
+@tsProps
 **/
 @withStyle(generateStyle, generateComponentTheme)
 @testable()
-class DrawerContent extends Component<
-  DrawerLayoutContentProps,
-  DrawerLayoutContentState
-> {
+class DrawerContent extends Component<DrawerLayoutContentProps> {
   static readonly componentId = 'DrawerLayout.Content'
 
   static locatorAttribute = 'data-drawer-content'
   static propTypes = propTypes
   static allowedProps = allowedProps
   static defaultProps = {
-    children: null,
-    // @ts-expect-error ts-migrate(6133) FIXME: 'el' is declared but its value is never read.
-    contentRef: (el: any) => {},
-    // @ts-expect-error ts-migrate(6133) FIXME: 'node' is declared but its value is never read.
-    onSizeChange: (node: any) => {},
     role: 'region'
   }
 
-  state: DrawerLayoutContentState = {
-    shouldTransition: false
-  }
+  ref: HTMLDivElement | null = null
 
-  ref: Element | null = null
   get _content() {
     console.warn(
       '_content property is deprecated and will be removed in v9, please use ref instead'
     )
 
     return this.ref
   }
-  _resizeListener = null
 
-  _debounced = null
-  _timeouts = []
+  private _resizeListener?: ResizeObserver
+
+  private _debounced?: Debounced
+  private _timeouts = []
```

Comment:
this is also unused,its always an empty array

Reply:
Oh, nice catch, haven't noticed! Thanks!